samples/Sentry.Samples.Console.Basic/Program.cs

```diff
@@ -33,66 +36,133 @@
+    transaction.Finish();
 
-        // Throw an exception
-        throw new ApplicationException("Something happened!");
+    async Task FirstFunction()
+    {
+        // This is an example of making an HttpRequest. A trace us automatically captured by Sentry for this.
+        var messageHandler = new SentryHttpMessageHandler();
+        var httpClient = new HttpClient(messageHandler, true);
+        var html = await httpClient.GetStringAsync("https://example.com/");
+        Console.WriteLine(html);
     }
-    catch (Exception exception)
+
+    async Task SecondFunction()
     {
-        // This is an example of capturing a handled exception.
-        SentrySdk.CaptureException(exception);
-        span.Finish(exception);
+        var span = transaction.StartChild("function", nameof(SecondFunction));
+
+        try
+        {
+            // Simulate doing some work
+            await Task.Delay(100);
+
+            // Throw an exception
+            throw new ApplicationException("Something happened!");
+        }
+        catch (Exception exception)
+        {
+            // This is an example of capturing a handled exception.
+            SentrySdk.CaptureException(exception);
+            span.Finish(exception);
+        }
+
+        span.Finish();
     }
 
-    span.Finish();
+    async Task ThirdFunction()
+    {
+        var span = transaction.StartChild("function", nameof(ThirdFunction));
+        try
+        {
+            // Simulate doing some work
+            await Task.Delay(100);
+
+            // This is an example of an unhandled exception.  It will be captured automatically.
+            throw new InvalidOperationException("Something happened that crashed the app!");
+        }
+        finally
+        {
+            span.Finish();
+        }
+    }
 }
 
-async Task ThirdFunction()
+// Heap dumps are only supported in .NET 6.0 or later.
+#if NET6_0_OR_GREATER
+async Task HeapDumpDemo(CancellationToken cancellationToken)
 {
-    var span = transaction.StartChild("function", nameof(ThirdFunction));
-    try
+    var memoryHog = new List<byte[]>();
+    Console.WriteLine();
+    Console.WriteLine("Hogging memory...");
+
+    // Sentry checks memory usage every time a full garbage collection occurs. It might take a while to trigger this,
+    // although we've configured some ridiculously aggressive settings in the runtimeconfig.template.json file to make
+    // this happen more quickly, for the purposes of this demo... definitely don't do this in production!
+    while (cancellationToken.IsCancellationRequested == false)
     {
-        // Simulate doing some work
-        await Task.Delay(100);
-
-        // This is an example of an unhandled exception.  It will be captured automatically.
-        throw new InvalidOperationException("Something happened that crashed the app!");
-    }
-    finally
-    {
-        span.Finish();
+        var array = new byte[2_000_000_000];
```

Comment:
We'd have to do it in a different sample since this one sets `<PublishAot>true</PublishAot>`... i.e. no reflection. I could create a separate demo specifically for heap dumps. Worth it?